Thecarisma-syntax.md

```diff
@@ -0,0 +1,620 @@
+
+    ```kotlin
+    func outer() {
+        var id = "JohnDoe"
+
+        func inner() {
+            var id = 1234568
+            print(id) //1234568
+        }
+        inner()
+        print(id) //"JohnDoe"
+    }
+
+    ```
+
+- A constant subject cannot be overshadowed
+
+    ```kotlin
+    func outer() {
+        const id = "JohnDoe"
+
+        func inner() {
+            var id = 1234568 //Invalid. fails
+            ...
+        }
+        ...
+    }
+
+    ```
+
+## IDENTIFIERS
+- Identifiers must start with a character and can be followed by characters, digits or underscore.
+
+- For now, characters can only be in the ASCII range. Support for certain UTF-8 characters is planned.
+
+- An example of a subject identifier
+
+    ```js
+    const name = "John Lennon"
+    ```
+
+- An example of a function identifier. More on functions [later](#functions).
+
+    ```swift
+    func get_number() -> Int {
+        return 20
+    }
+    ```
+
+- An example of a type identifier. More on types [later](#types).
+
+    ```swift
+    type Person {
+        name: String,
+        age: Int,
+    }
+    ```
+
+## SEMI COLONS
+- Semi colons are used to separate multiple statements or expressions.
+
+    ```swift
+    const name = "John"; const age = 40;
+    print(name, age);
+    ```
+
+- Semi colons are optional if statements/expressions are separated by newlines.
+
+    ```cpp
+    const name = "John"
+    const age = 40
+    print(name, age)
+    ```
+
+## TYPE ANNOTATION
+- Types can be automatically inferred in Ratio.
+    ```js
+    var todo = "Create a new programming language :)" // `todo` is inferred as `String`
+    ```
+
+    ```cpp
+    func add(a, b) {
+        return a + b
+    }
+
+    const sum = add(1, 1) // Usage of `add` inferred as `(Int, Int) -> Int` here.
+    ```
+
+- Subjects can be explicitly type-annotated.
+
+    ```js
+    var identification: Int = 60
+    ```
+
+- Functions can also be explicitly type-annotated.
+
+    ```swift
+    func add(a: Int, b: Int) -> Int {
+        return a + b
+    }
+    ```
+
+
+## DESTRUCTURING
+- Destructuring a tuple.
+
+    ```js
+    const (name, age) = ("John", 50)
+    ```
+
+- Destructuring a list.
+
+    ```js
+    const [..., last] = [1, 2, 3, 4, 5]
+    ```
+
+- Destructuring an object.
+
+    ```js
+    const {name, age} = john
+    ```
+
+## IF EXPRESSION
+- Conditional branches using `if`, `else if` and `else`.
+
+    ```swift
+    if cond1 {
+        print("if")
+    } else if cond2 {
+        print("else if")
+    } else {
+        print("else ")
+    }
+    ```
+
+- Checking for falsy values.
+
+    ```swift
+    if !complete { 
+        print("incomplete")
+    }
+    ```
+
+- Checking for Noneness and binding value to a name.
+
+    ```swift
+    if const stock_code = get_stock_code("") {
+        print(stock_code)
+    }
+    ```
+
+## FOR LOOP
+- A `for` loop iterates through an iterable and binds the value to a subject.
+
+    ```py
+    for i in 1:11 {
```

Comment:
Is this for 1 to 10 (exclusive of 11)? 

A quick suggestion, how about having a syntax for the last number included e.g. `:=`?

I believe having an inclusive range syntax would improve the way people would write (instead of +1 every single time).

Reply:
Also, how does on include a **step size** for the loop ? We can't afford to make including a **step size** arg to a for loop with <lower bound>:<upper bound> as cumbersome as it is currently in python 3 if you ask me.